docs/TESTING.md

```diff
@@ -45,3 +45,29 @@ make allnoconfig
 make Image
 head -c 4096 arch/arm64/boot/Image > test_image.bin
 ```
+
+### Generating `ELF` images with [ELFIO](https://github.com/serge1/ELFIO)
+
+All the ELF files used in `linux-loader` are generated using the `ELFIO` tool.
+The [`.cpp` source files](elfio_files) are created from the ELFIO `writer`
+example, with minimal changes on top.
+
+| Source File | Generated Binary File |
+|-------------|-----------------------|
+| bad_align_writer.cpp | test_bad_align.bin |
+
+#### Example for generating `test_bad_align.bin`
+
+```bash
+git clone git@github.com:serge1/ELFIO.git
+
+# Copy the bad_align_writer.cpp file from this repo to the ELFIO path.
+cp ${LINUX_LOADER_PATH}/docs/elfio/overflow_elfnote.cpp ${ELFIO_PATH/
```

Comment:
```suggestion
cp ${LINUX_LOADER_PATH}/docs/elfio/overflow_elfnote.cpp ${ELFIO_PATH}/
```

I'd also enclose the paths within double quotes, particularly because you dereference variables, but that's just a style nit.

Reply:
Also, there's no `overflow_elfnote.cpp`. Was it supposed to be `bad_align_writer.cpp`?

Reply:
And the intermediary directory is `elfio_files`, not `elfio`.

Reply:
Looks like this is indeed outdated. I'll update it.

Reply:
Updated.